progressist/__init__.py

```diff
@@ -115,12 +115,28 @@ def speed(self):
         """Number of iterations per second."""
         return Float(1.0 / self.avg if self.avg else 0)
 
+    @property
+    def throttled(self):
+        if not self.throttle:
+            return
+        if isinstance(self.throttle, (int, float)):
+            throttle = self.throttle
+            if isinstance(self.throttle, float):
+                throttle = max(1, self.total * self.throttle)
+            throttle = self._last_render + throttle
+            if self.done < throttle:
+                if not self.total or throttle <= self.total:
```

Comment:
So if we haven't reached the throttle (`self.done < throttle`), but the throttle is higher than the total, you seem to be considering that we should print. Why?

To me, if the throttle is higher than the total and we haven't reached it, then you need to check whether we've reached the total or not.

If we haven't reached the total, then you should not print, because if the throttle is higher than the total, then the next time you print is when you reach the total.

I do think this would all get clearer if you stored the `_next_render` instead of the `_last_render` value. `next_render` would be equal to `min(self._last_render + throttle, self.total)`

This way you never have a `_next_render` further than the total, and the comparison with `self.done` becomes easier?